File: php/features.php
<?php
declare(strict_types=1);

// AddressLookupService SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AddressLookupServiceFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AddressLookupServiceBaseFeature();
            case "test":
                return new AddressLookupServiceTestFeature();
            default:
                return new AddressLookupServiceBaseFeature();
        }
    }
}

File: py/addresslookupservice_sdk/features.py
# AddressLookupService SDK feature factory

from addresslookupservice_sdk.feature.base_feature import AddressLookupServiceBaseFeature
from addresslookupservice_sdk.feature.test_feature import AddressLookupServiceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AddressLookupServiceBaseFeature(),
        "test": lambda: AddressLookupServiceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
